2005-12-12  Gwenole Beauchesne  <gbeauchesne@mandriva.com>

	* checkboxtree.c (newtCheckboxTreeAddItem): Remove va_start()
	duplicates.

	* grid.c (newtGridHCloseStacked): Use va_end() when we have
	processed the varargs.
	(newtGridVCloseStacked): Likewise.
	(newtGridVStacked): Likewise.
	(newtGridHStacked): Likewise.

	* form.c (formOps): Make it local to the DSO.
	* grid.c (formOps): Fix declaration.

--- newt-0.51.6/grid.c.assorted-fixes	2002-08-15 16:13:40.000000000 -0400
+++ newt-0.51.6/grid.c	2005-12-12 09:18:23.000000000 -0500
@@ -26,8 +26,8 @@ struct grid_s {
     struct gridField ** fields;
 };
 
-/* this is a bit of a hack */
-extern struct componentOps formOps[];
+/* make it local to the dso */
+extern struct componentOps formOps __attribute__ ((visibility ("hidden")));
 
 newtGrid newtCreateGrid(int cols, int rows) {
     newtGrid grid;
@@ -108,7 +108,7 @@ static void shuffleGrid(newtGrid grid, i
 		    shuffleGrid(field->u.grid, left, top, 0);
 		j = field->u.grid->width;
 	    } else if (field->type == NEWT_GRID_COMPONENT) {
-		if (field->u.co->ops == formOps)
+		if (field->u.co->ops == &formOps)
 		    newtFormSetSize(field->u.co);
 		j = field->u.co->width;
 	    } else 
@@ -319,7 +319,7 @@ newtGrid newtGridHCloseStacked(enum newt
 
     grid = stackem(0, type1, what1, args, 1);
 
-    va_start(args, what1);
+    va_end(args);
 
     return grid;
 }
@@ -332,7 +332,7 @@ newtGrid newtGridVCloseStacked(enum newt
 
     grid = stackem(1, type1, what1, args, 1);
 
-    va_start(args, what1);
+    va_end(args);
 
     return grid;
 }
@@ -345,7 +345,7 @@ newtGrid newtGridVStacked(enum newtGridE
 
     grid = stackem(1, type1, what1, args, 0);
 
-    va_start(args, what1);
+    va_end(args);
 
     return grid;
 }
@@ -358,7 +358,7 @@ newtGrid newtGridHStacked(enum newtGridE
 
     grid = stackem(0, type1, what1, args, 0);
 
-    va_start(args, what1);
+    va_end(args);
 
     return grid;
 }
--- newt-0.51.6/checkboxtree.c.assorted-fixes	2003-01-03 15:54:45.000000000 -0500
+++ newt-0.51.6/checkboxtree.c	2005-12-12 09:15:56.000000000 -0500
@@ -115,7 +115,6 @@ int newtCheckboxTreeAddItem(newtComponen
     va_start(argList, index);
     numIndexes = 0;
     i = index;
-    va_start(argList, index);
     while (i != NEWT_ARG_LAST) {
 	indexes[numIndexes++] = i;
 	i = va_arg(argList, int);
--- newt-0.51.6/form.c.assorted-fixes	2005-12-12 09:13:21.000000000 -0500
+++ newt-0.51.6/form.c	2005-12-12 09:19:26.000000000 -0500
@@ -415,7 +415,7 @@ static void formPlace(newtComponent co, 
 static newtCallback helpCallback;
 
 /* this isn't static as grid.c tests against it to find forms */
-struct componentOps formOps = {
+struct componentOps formOps __attribute__ ((visibility ("hidden"))) = {
     newtDrawForm,
     formEvent,
     newtFormDestroy,
